app/controllers/static_controller.rb

```diff
@@ -101,8 +101,7 @@ def enter
            (uri.host.blank? || uri.host == forum_uri.host) &&
            uri.path !~ /\./
 
-          destination = uri.path
-          destination = "#{uri.path}?#{uri.query}" if uri.path =~ /new-topic/ || uri.path =~ /new-message/ || uri.path =~ /user-api-key/
+          destination = "#{uri.path}#{uri.query ? "?#{uri.query}" : ""}"
```

Comment:
This does seem slightly less secure than before. However, I can't think of a way `uri.query` could be exploited so it's probably fine.